core/miner/storage_fsm/impl/precommit_batcher_impl.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "miner/storage_fsm/impl/precommit_batcher_impl.hpp"
+
+#include <utility>
+
+namespace fc::mining {
+
+  PreCommitBatcherImpl::PreCommitBatcherImpl(size_t maxTime,
+                                             std::shared_ptr<FullNodeApi> api)
+      : api_(std::move(api)){};
+
+  PreCommitBatcherImpl::preCommitEntry::preCommitEntry(
+      primitives::TokenAmount number, api::SectorPreCommitInfo info)
+      : deposit(std::move(number)), pci(std::move(info)){};
+
+  PreCommitBatcherImpl::preCommitEntry &
+  PreCommitBatcherImpl::preCommitEntry::operator=(const preCommitEntry &x) =
+      default;
+
+  outcome::result<std::shared_ptr<PreCommitBatcherImpl>>
+  PreCommitBatcherImpl::makeBatcher(
+      size_t maxWait,
+      std::shared_ptr<FullNodeApi> api,
+      std::shared_ptr<libp2p::protocol::Scheduler> scheduler,
+      Address &miner_address) {
+    struct make_unique_enabler : public PreCommitBatcherImpl {
+      make_unique_enabler(size_t MaxTime, std::shared_ptr<FullNodeApi> api)
+          : PreCommitBatcherImpl(MaxTime, std::move(api)){};
+    };
+
+    std::shared_ptr<PreCommitBatcherImpl> batcher =
+        std::make_unique<make_unique_enabler>(maxWait, std::move(api));
+
+    batcher->miner_address_ = miner_address;
+    batcher->maxDelay = maxWait;
+
+    batcher->cutoffStart = std::chrono::system_clock::now();
+    batcher->closestCutoff = maxWait;
+    batcher->handle_ = scheduler->schedule(maxWait, [=]() {
+      auto maybe_send = batcher->sendBatch();
+      batcher->handle_.reschedule(maxWait);  // TODO: maybe in gsl::finally
+    });
+    return batcher;
+  }
+
+  outcome::result<void> PreCommitBatcherImpl::sendBatch() {
+    std::unique_lock<std::mutex> locker(mutex_, std::defer_lock);
+
+    OUTCOME_TRY(head, api_->ChainHead());
+    OUTCOME_TRY(minfo, api_->StateMinerInfo(miner_address_, head->key));
+
+    PreCommitBatch::Params params = {};
+    for (auto &data : batchStorage) {
+      mutualDeposit += data.second.deposit;
+      params.sectors.push_back(data.second.pci);
+    }
+    // TODO: goodFunds = mutualDeposit + MaxFee; -  for checking payable
+    if (params.sectors.size() != 0) {
+      OUTCOME_TRY(encodedParams, codec::cbor::encode(params));
+
+      auto maybe_signed = api_->MpoolPushMessage(
+          vm::message::UnsignedMessage(
+              miner_address_,
+              minfo.worker,  // TODO: handle worker
+              0,
+              mutualDeposit,
+              {},
+              {},
+              vm::actor::builtin::v0::miner::PreCommitSector::Number,
+              MethodParams{encodedParams}),
+          kPushNoSpec);
+
+      if (maybe_signed.has_error()) {
+        return maybe_signed.error(); //TODO: maybe logs
+      }
```

Comment:
Change the code above and remove this check
```suggestion
      OUTCOME_TRY(maybe_signed, api_->MpoolPushMessage(...))
```

Reply:
outcome_try are not helpful in delayed function, I ve decided to use logs to observe the behaviour

Reply:
fixed